api/cluster.go

```diff
@@ -303,12 +304,24 @@ func (s *Server) localTagDelByQuery(request models.IndexTagDelByQuery) (int, err
 		return 0, err
 	}
 
-	if request.Execute {
+	if !request.Execute {
+		// Use FindTerms to count what would be deleted
+		matched, _ := s.MetricIndex.FindTerms(request.OrgId, []string{}, query)
+		return int(matched), nil
+	}
+
+	if request.Method == "archive" {
+		casidx, ok := s.MetricIndex.(*cassandra.CasIdx)
+		if !ok {
+			return 0, errors.New("Only cassandra index supports 'archive' method")
+		}
+		deleted, err := casidx.ArchiveTagged(request.OrgId, query)
+		return len(deleted), err
+	} else if request.Method == "delete" {
 		deleted, err := s.MetricIndex.DeleteTagged(request.OrgId, query)
 		return len(deleted), err
 	}
-	matched, _ := s.MetricIndex.FindTerms(request.OrgId, []string{}, query)
-	return int(matched), nil
+	return 0, fmt.Errorf("Unknown delete method '%s'", request.Method)
```

Comment:
pretty sure the model validation already takes care of this case. but anyway, doesn't hurt.

Reply:
Model validation takes care of this when this code and the validation code agree. This is just defensive handling in case these code sections accidentally get out of sync.